base/instruction.py

```diff
@@ -1783,24 +1794,44 @@ def symbols(self):
             r, _ = self
             yield r
 
+    class trap(interface.namedtypedtuple, interface.symbol_t):
+        """
+        A tuple for representing a trap code that can be encoded within
+        certain instructions on the MIPS architecture.
+
+        Simply wraps the encoded integer as a individual value in a tuple
+        with the format `(code)`.
+        """
+        _fields = ('code',)
+        _types = (six.integer_types,)
+
+        code = property(fget=operator.itemgetter(0))
+
+        @property
+        def symbols(self):
+            '''This operand type is not composed of any symbols.'''
+            raise StopIteration
+            yield   # so that this function is still treated as a generator
+
+    @staticmethod
+    def coprocessor(regnum):
+        """
+        A callable that returns a co-processor register for the MIPS architecture.
+
+        Takes a `regnum` argument which returns the correct register.
+        """
+        global architecture
+        return architecture.by_coprocessor(regnum)
+
     @staticmethod
-    def coproc(regnum):
+    def float(regnum):
         """
         A callable that returns a co-processor for the MIPS architecture.
```

Comment:
The documentation for this could reference the coprocessor index. Also remove the "-" (hyphen) when describing it.